core/domain/suggestion_services.py

```diff
@@ -542,3 +556,65 @@ def check_can_resubmit_suggestion(suggestion_id, user_id):
     suggestion = get_suggestion_by_id(suggestion_id)
 
     return suggestion.author_id == user_id
+
+
+def check_can_edit_suggestion(user, suggestion_id):    # pylint: disable=too-many-return-statements
+    """Checks whether the user have permission to edit suggestion or not.
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        suggestion_id: str. Id of a suggestion.
+
+    Returns:
+        bool: Whether the user can edit the suggestion.
+    """
+
+    suggestion = get_suggestion_by_id(suggestion_id)
+    user_id = user.user_id
+    target_types = ['exploration', 'collection', 'question']
+
+    if suggestion is None:
+        raise Exception('Suggestion does not exists.')
+
+    if suggestion.is_handled:
+        raise Exception('The suggestion has already been accepted/rejected.')
+
+    if suggestion.author_id == user_id:
+        return True
+
+    if suggestion.target_type == suggestion_models.TARGET_TYPE_EXPLORATION:
+        exploration_rights = rights_manager.get_exploration_rights(
+            suggestion.target_id)
+        if exploration_rights is None:
+            return False
+
+        if rights_manager.check_can_edit_activity(
+                user, exploration_rights):
+            return True
+
+    elif suggestion.target_type == suggestion_models.TARGET_TYPE_COLLECTION:
+        collection_rights = rights_manager.get_collection_rights(
+            suggestion.target_id)
+        if collection_rights is None:
+            return False
+
+        if rights_manager.check_can_edit_activity(
+                user, collection_rights):
+            return True
+
+    elif suggestion.target_type == suggestion_models.TARGET_TYPE_QUESTION:
+        question_rights = question_services.get_question_rights(
+            suggestion.target_id, strict=False)
+        if question_rights is None:
+            return False
+
+        if (
+                role_services.ACTION_EDIT_ANY_QUESTION in user.actions or
+                question_rights.is_creator(user.user_id)):
+            return True
+
+    elif suggestion.target_type not in target_types:
+        raise Exception('Invalid suggestion target type.')
+
```

Comment:
We are missing lots of checks here. (Please go through the acl_dcorator for accepting/rejecting the suggestion. (I think you are missing the user's score and whether their role allows them to edit any activity etc.)

Reply:
ok @DubeySandeep but can you tell me what exactly is the user's score

Reply:
Have you gone through the acl for accepting/rejecting suggestion? (I think you'll understand once you'll go through it, let me know if it doesn't helps.)

Reply:
thanks, @DubeySandeep for pointing this, but can you tell me to how to test this functionality.
Currently, after adding this, I got an error "You do not have credentials to edit this suggestion." so can you tell me how to increase user score above the threshold. And also tell about what is the threshold value.
Thanks!!

Reply:
I'm not sure, you'll have to go through the existing code and see how it works!